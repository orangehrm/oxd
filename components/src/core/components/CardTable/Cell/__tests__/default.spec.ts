import {mount} from '@vue/test-utils';
import DefaultCell from '@orangehrm/oxd/core/components/CardTable/Cell/Default.vue';

const GLOBAL = {
  provide: {
    tableProps() {
      return {
        items: [],
      };
    },
  },
};

describe('CardTable > Cell > Default.vue', () => {
  it('should renders OXD CardTable > Cell > Default', () => {
    const wrapper = mount(DefaultCell, {
      global: GLOBAL,
      props: {item: 'Test'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD CardTable > Cell > Default with number', () => {
    const wrapper = mount(DefaultCell, {
      global: GLOBAL,
      props: {item: 2},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD CardTable > Cell > Default with array', () => {
    const wrapper = mount(DefaultCell, {
      global: GLOBAL,
      props: {item: ['Test', 'Array']},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD CardTable > Cell > Default with object', () => {
    const wrapper = mount(DefaultCell, {
      global: GLOBAL,
      props: {item: {test: 'Object'}},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
