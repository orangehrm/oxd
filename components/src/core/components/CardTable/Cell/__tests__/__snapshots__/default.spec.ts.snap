// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable > Cell > Default.vue should renders OXD CardTable > Cell > Default 1`] = `"Test"`;

exports[`CardTable > Cell > Default.vue should renders OXD CardTable > Cell > Default with array 1`] = `
"[
  \\"Test\\",
  \\"Array\\"
]"
`;

exports[`CardTable > Cell > Default.vue should renders OXD CardTable > Cell > Default with number 1`] = `"2"`;

exports[`CardTable > Cell > Default.vue should renders OXD CardTable > Cell > Default with object 1`] = `
"{
  \\"test\\": \\"Object\\"
}"
`;
